bin/jruby.sh

```diff
@@ -511,25 +511,52 @@ add_log "Detected Java version: $java_version"
 # Split version out for integer comparisons
 java_major=${java_version%%.*}
 
-# AppCDS support
-if [ "$java_major" -ge 13 ] && exists "$JAVA_HOME"/lib/server/*.jsa; then
-    java_has_appcds=true
+# Java version-specific tuning flags
+if [ "$java_major" -ge 13 ]; then
+    if exists "$JAVA_HOME"/lib/server/*.jsa; then
+        # AppCDS support
+        java_has_appcds=true
+    else
+        java_has_appcds=false
+    fi
+
+    if [ "$java_major" -ge 19 ]; then
+        # AppCDS autogeneration
+        java_has_appcds_autogenerate=true
+
+        if [ " $java_major" -ge 22 ]; then
+            # Native access
+            enable_native_access=true
+
+            if [ " $java_major" -ge 23 ]; then
```

Comment:
You've got spaces in your quoted variables

Reply:
Fixed